### PR TITLE
Add autofix to function-calc-no-unspaced-operator

### PR DESCRIPTION
**Pull request by doing-art**
> Which issue, if any, is this issue related to?

This PR contains implementation of the autofix for the `function-calc-no-unspaced-operator` rule as it is required in #4866.

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by doing-art** [comment]
Hi all
Could somebody please trigger a rebuild on the CI build? One check failed and it seems to be happened because of some infrastructural issue.
Thank you

File: lib/rules/function-calc-no-unspaced-operator/index.js [hunk 1]
```diff
@@ -108,15 +137,69 @@ function rule(actual) {
...
 						}
 					});
 				}
 			});
+
+			if (context.fix) {
+				decl.value = symbolsToFix.reduce((fixedValue, { insert, index }) => {
+					shiftIndexes(symbolsToFix, index, insert);
+
+					return insert ? insertSpace(fixedValue, index) : removeSpace(fixedValue, index);
```
Comment (ybiquitous) [inline lib/rules/function-calc-no-unspaced-operator/index.js:180]:
```suggestion
					return insert
						? insertCharAtIndex(fixedValue, index, ' ')
						: removeCharAtIndex(fixedValue, index);
```

File: lib/rules/function-calc-no-unspaced-operator/index.js [hunk 2]
```diff
@@ -108,15 +137,69 @@ function rule(actual) {
...
+	});
+}
+
+function removeSpace(value, index) {
+	return value.slice(0, index) + value.slice(index + 1, value.length);
+}
+
+function insertSpace(value, index) {
+	return `${value.substring(0, index)} ${value.substr(index)}`;
+}
```
Comment (ybiquitous) [inline lib/rules/function-calc-no-unspaced-operator/index.js:201]:
```suggestion
function removeCharAtIndex(str, index) {
	return str.slice(0, index) + str.slice(index + 1, str.length);
}

function insertCharAtIndex(str, index, char) {
	return str.slice(0, index) + char + str.slice(index, str.length);
}
```

**Review by ybiquitous — COMMENTED** [review]
[suggestion] How about using more generic function names and unifying the use of `String.prototype.slice()` instead of `substr()` or `substring()`?

IMO, `substr()` and `substring()` seem confusing, so I recommend using `slice()` instead.

See also:
- http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr
- https://github.com/sindresorhus/eslint-plugin-unicorn/blob/694934f96c1967a646642d791052f0450489f902/docs/rules/prefer-string-slice.md

**Comment by doing-art** [comment]
> @doing-art Thank you for creating this change! 👍🏼
> 
> The logic seems good to me, but I've left some comments to improve the code readability and maintainability.

Thank you for reviewing my PR and sharing knowledge about `String` methods!
I completely agree with all your comments. The PR is updated as you have suggested. Please take a look at it when you get a chance.

**Comment by ybiquitous** [comment]
@doing-art Thank you! It seems good to me. 👍🏼 

By the way, we are now working on the next version 14 (#5205), which includes many significant changes such as ESM, PostCSS 8, etc.
So, I think it seems good to change this PR's base branch from `mater` to [`v14`](https://github.com/stylelint/stylelint/tree/v14).

What do you think? @jeddy3

**Comment by jeddy3** [comment]
> So, I think it seems good to change this PR's base branch from mater to v14.

I agree. We'll want to minimise conflicts and save ourselves a headache by working from the `v14` branch.

@doing-art Can you change the merge branch, please?

**Comment by doing-art** [comment]
@jeddy3 @ybiquitous Changed the merge branch and resolved one small conflict in the tests.

**Comment by jeddy3** [comment]
> Changed the merge branch and resolved one small conflict in the tests.

Thanks! 

You'll need to run `npm run format` to fix formatting. Tests are looking good, though.

**Comment by doing-art** [comment]
> > Changed the merge branch and resolved one small conflict in the tests.
> 
> Thanks!
> 
> You'll need to run `npm run format` to fix formatting. Tests are looking good, though.

Sorry, my bad. Have fixed linting issue.

**Review by ybiquitous — APPROVED** [review]
@doing-art Thank you! Looks good to me. 👍🏼

**Review by jeddy3 — APPROVED** [review]
LGTM, thanks!

**Comment by jeddy3** [comment]
`v14` changelog updated:

> - Added: `function-calc-no-unspaced-operator` autofix ([#5273](https://github.com/stylelint/stylelint/pull/5273)).

**Pull request merged**